backend/Services/CoordinateFetcher.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Services;
+
+use GuzzleHttp\Client;
+use Internships\Models\FetchAddress;
+
+class CoordinateFetcher
+{
+    protected string $mapboxToken;
+
+    public function __construct()
+    {
+        $this->mapboxToken = $_ENV["MAPBOX_TOKEN"];
+    }
+
+    public function coordinatesFromAddress(string $address = "", FetchAddress $addressObject = null): string
+    {
+        if ($addressObject) {
+            $address = $this->fetchAddressToString($addressObject);
+        }
+
+        $api = new Client();
+        $addressUrl = urlencode($address);
+
+        $url = "https://api.mapbox.com/geocoding/v5/mapbox.places/{$addressUrl}.json"
+            . "?access_token={$this->mapboxToken}"
+            . "&limit=1";
+
+        $response = $api->get($url);
+        $content = json_decode($response->getBody()->getContents(), true);
+        $coordinates = $content["features"][0]["geometry"]["coordinates"];
```

Comment:
Maybe use `try {} catch()` to avoid unexpected errors?

Reply:
There's try catch statement above on function call. I can see that in some cases I want program to throw error, in other just skip that.

I could add custom exception to replace generic 'not found' message there and catch new throw in caller function, but I think I'll change the function to modify coordinates through argument, and return boolean based on success. Final result should look better.

I need to see how it will turn out, so I might push one variant or another.

Reply:
I settled on approach with boolean returning, it eased 'staircases' in other part of the code.